server/controllers/userController.js

```diff
@@ -97,10 +97,41 @@ const register = async (req, res) => {
   }
 };
 
+const login = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    // check if user exists
+    const findUser = await findUserByEmail(email);
+
+    if (!findUser) {
+      return res.json({ error: 'Incorrect Email or Password' });
+    }
+    // Load hash, compare password
+    const hash = await findUser.password;
+    const pwMatch = await bcrypt.compare(password, hash);
+
+    if (!pwMatch) {
+      return res.json({ error: 'Incorrect Email or Password' });
+    }
+
+    // generate JWT
+    const payload = { email };
+    const token = jwt.sign(payload, process.env.JWTSECRET, {
+      expiresIn: '1h',
+    });
+
+    return res.cookie('token', token, { httpOnly: true }).send({ email }); // maybe favorites
+  } catch (err) {
+    return res.json({ err });
```

Comment:
I believe this is to catch async / await errors, yeah? These should be handled through our global error handler located in some file probably in the middleware folder. You can call next with an error object and express will automatically funnel your error to that error handler. Check that file to ensure you include the right properties in your error and that we're not giving the client too much information about what this error from our database or server logic might be.